atom/browser/api/atom_api_app.cc

```diff
@@ -262,6 +263,36 @@ v8::Local<v8::Value> App::DefaultSession(v8::Isolate* isolate) {
     return v8::Local<v8::Value>::New(isolate, default_session_);
 }
 
+bool App::OnProcessSingletonNotification(
+    const base::CommandLine& command_line,
+    const base::FilePath& current_directory) {
+  ProcessSingleton::NotificationCallback cb;
+  mate::Converter<ProcessSingleton::NotificationCallback>::FromV8(isolate(), single_instance_callback_, &cb);
+  
+  return cb.Run(command_line, current_directory);
+}
+
+bool App::MakeSingleInstance(v8::Local<v8::Function> callback) {
```

Comment:
@zcbenz @deepak1556 @hokein I'm super close to making this work but I'm messing up trying to marshal this callback correctly - basically, we need to take this callback, convert it to a `ProcessSingleton::NotificationCallback`, then hold it in such a way so that the V8 function doesn't go out of scope (oh, and probably call it on the right thread). Help!